extensions/wikia/WallNotifications/WallNotifications.class.php

```diff
@@ -387,50 +387,65 @@ protected function sendEmails( array $watchers, WallNotificationEntity $notifica
 			$watcher = $this->getUser( $val );
 			$mode = $watcher->getOption( 'enotifwallthread' );
 
-			if ( !empty( $mode ) && $watcher->getId() != 0 && (
+			if ( !( $watcher->getBoolOption('unsubscribed') === true ) &&
+				!empty( $mode ) && $watcher->getId() != 0 && (
 				( $mode == WALL_EMAIL_EVERY ) ||
 				( $mode == WALL_EMAIL_SINCEVISITED && empty( $this->uniqueUsers[$entityKey][$watcher->getId()] ) )
 			)) {
 
-				$key = $this->createKeyForMailNotification( $watcher->getId(), $notification );
-				$watcherName = $watcher->getName();
-
-				if ( $notification->data->msg_author_username == $notification->data->msg_author_displayname ) {
-					$author_signature = $notification->data->msg_author_username;
+				$controller = $this->getEmailExtensionController( $notification );
```

Comment:
Ah, I see. I thought of this as binary, we either were sending a Forum email or a Wall email, but if we're differentiating those emails even further then having this method makes more sense to me. Leaving as is works for me :+1: